### DISCUSSION TITLE
Stream audiobooks without encryption

### DISCUSSION THREAD

**Discussion by Mbucari** [mkb79/Audible#61] [Ideas]
I discovered that if you make a licenserequest with consumption_type=Streaming and drm_type=Dash, Audible will serve up unencrypted mpeg-4a Dash stream. The Dash mpd url is in content_license.license_response, and it doesn't require an authenticated request to stream. Just open that url in VLC and it plays. It works for both purchased and Audible Plus content.

I'm not sure what to do with this information, but it seems like it may make a nice new feature for audible-cli. You could just retrieve the mpd url for the user to use, you could download all pieces into a folder, or, you could download all pieces and combine all movie fragment boxes into a single mpeg-4 file.

**Comment by mkb79** [comment]
Hi. Thank you for this information. You are know which quality the stream haves?
Have you knowledge in web design? I'm working on an Django app for Audible.

EDIT:
You are know, if the mpd URL have a specific lifetime?

**Reply by Mbucari** [reply]
> Hi. Thank you for this information. You are know which quality the stream haves?

The same qualities that are offered with consumption_type=Download: Low, Normal, High and Extreme, but only two different audio qualities are served.

> Have you knowledge in web design? I'm working on an Django app for Audible.

Very little. I've played with asp.net and SQL, but I don't have even basic competence in web development.

**Reply by Mbucari** [reply]
> You are know, if the mpd URL have a specific lifetime?

I don't fully understand how it works. I did some testing and found that the link can only be accessed for a couple of minutes after the license request. But if you start streaming before the link expires, it appears that it does not expire as long as you keep streaming. I tested an audiobook for 15 minutes in VLC, and it did not stop playing.

**Reply by mkb79** [reply]
I will implement this in audible-cli and my new Django project.

**Comment by mkb79** [comment]
And thank you very much for sharing your information!

**Reply by Mbucari** [reply]
I tested this again recently and was unable to make it work.  It was likely a bug, and they patched it.

**Reply by mkb79** [reply]
Oh, that sounds not so good. But thank you for this information.
